hibernate-reactive-core/src/test/java/org/hibernate/reactive/SchemaUpdateSqlServerTestBase.java

```diff
@@ -0,0 +1,340 @@
+
+		@Override
+		protected Configuration constructConfiguration(String hbm2DdlOption) {
+			final Configuration configuration = super.constructConfiguration( hbm2DdlOption );
+			configuration.setProperty( Settings.HBM2DDL_JDBC_METADATA_EXTRACTOR_STRATEGY, INDIVIDUALLY.toString() );
+			return configuration;
+		}
+	}
+
+	public static class GroupedSchemaUpdateSqlServerTestBase extends SchemaUpdateSqlServerTestBase {
+
+		@Override
+		protected Configuration constructConfiguration(String hbm2DdlOption) {
+			final Configuration configuration = super.constructConfiguration( hbm2DdlOption );
+			configuration.setProperty( Settings.HBM2DDL_JDBC_METADATA_EXTRACTOR_STRATEGY, GROUPED.toString() );
+			return configuration;
+		}
+	}
+
+	protected Configuration constructConfiguration(String hbm2DdlOption) {
+		Configuration configuration = constructConfiguration();
+		configuration.setProperty( Settings.HBM2DDL_AUTO, hbm2DdlOption );
+		// DEFAULT_CATALOG needs to be set to something other than an empty string;
+		// otherwise, NullPointerException is thrown.
+		configuration.setProperty( Settings.DEFAULT_CATALOG, "master" );
+		configuration.setProperty( Settings.DEFAULT_SCHEMA, "dbo" );
+		configuration.setProperty( Settings.SHOW_SQL, "true" );
+		return configuration;
+	}
+
+	@Rule
+	public DatabaseSelectionRule dbRule = DatabaseSelectionRule.runOnlyFor( SQLSERVER );
+
+	@Before
+	@Override
+	public void before(TestContext context) {
+		Configuration createHbm2ddlConf = constructConfiguration( "create" );
+		createHbm2ddlConf.addAnnotatedClass( ASimpleFirst.class );
+		createHbm2ddlConf.addAnnotatedClass( AOther.class );
+
+		test( context, setupSessionFactory( createHbm2ddlConf )
+				.thenCompose( v -> factoryManager.stop() ) );
+	}
+
+	@After
+	@Override
+	public void after(TestContext context) {
+		final Configuration dropHbm2ddlConf = constructConfiguration( "drop" );
+		dropHbm2ddlConf.addAnnotatedClass( ASimpleNext.class );
+		dropHbm2ddlConf.addAnnotatedClass( AOther.class );
+		dropHbm2ddlConf.addAnnotatedClass( AAnother.class );
+
+		test( context, factoryManager.stop()
+				.thenCompose( v -> setupSessionFactory( dropHbm2ddlConf ) )
+				.thenCompose( v -> factoryManager.stop() ) );
+	}
+
+	@Test
+	public void testValidationSucceed(TestContext context) {
+		Configuration createHbm2ddlConf = constructConfiguration( "validate" );
+		createHbm2ddlConf.addAnnotatedClass( ASimpleFirst.class );
+		createHbm2ddlConf.addAnnotatedClass( AOther.class );
+
+		test( context, setupSessionFactory( createHbm2ddlConf ) );
+	}
+
+	//TODO: I'm just checking that the validation fails because the table is missing, but we need more tests to check that
+	//      it fails for other scenarios: missing column, wrong type (?) and so on. (I don't know exactly what cases `validate`
+	//      actually checks).
+	@Test
+	public void testValidationFails(TestContext context) {
+		Configuration createHbm2ddlConf = constructConfiguration( "validate" );
```

Comment:
I guess this should be called `validateHbm2ddlConf`

Reply:
I would prefer the generic configuration, as used in the other tests.

WDYT?